conans/model/options.py

```diff
@@ -139,6 +139,8 @@ def rm_safe(self, field):
             delattr(self, field)
         except ConanException:
             pass
+        except KeyError:
```

Comment:
This needs to be clearly explained, it might change existing behavior, would need a separate PR with tests.

Reply:
Yes, I agree this is a major change. In this case, because in the `_ensure_exists()` method there is a check when the options are `_constrained`. I am not sure what this means so I left this as it is right now and wait for feedback

Reply:
Lets make sure this PR is not merged until this is addressed, it is still not evident why it is necessary, if the functionality is basically what the recipes in ConanCenter are doing and ``xxx_safe()`` is used.